apps/api_web/lib/api_web/controllers/portal/session_controller.ex

```diff
@@ -35,6 +41,32 @@ defmodule ApiWeb.ClientPortal.SessionController do
     end
   end
 
+  def new_2fa(conn, _params) do
```

Comment:
I kind of feel that way about the entire admin session controller (that it's redundant). There's nothing stopping and admin from logging in the normal path and then navigating to an admin page or vice versa, so I don't really see the point in having an separate controller for admin login/logout, but that kind of change felt out of scope.

I can take a stab at consolidating the 2fa stuff into a shared controller.

Reply:
I've extracted MFA verification into it's own controller. I ended up putting it in it's own scope that's used by both login flows rather than using the controller in both login flows, since otherwise I don't have a good way of know what helper function to use when posting back the form (`mfa_path`, or `admin_mfa_path`).